tests/test_submission.py

```diff
@@ -54,3 +65,106 @@ def test_try_edit_locked_by_current_user_submission(
         method="patch", url=f"/api/metadata_submission/{submission.id}", json=json.loads(payload)
     )
     assert response.status_code == 200
+
+
+def test_submission_list_with_roles(db: Session, client: TestClient, token: Token, logged_in_user):
+    user_a = fakes.UserFactory()
+    submission_a = fakes.MetadataSubmissionFactory(author=user_a, author_orcid=user_a.orcid)
+    submission_b = fakes.MetadataSubmissionFactory(
+        author=logged_in_user, author_orcid=logged_in_user.orcid
+    )
+    fakes.MetadataSubmissionFactory(author=user_a, author_orcid=user_a.orcid)
+    db.commit()
+    fakes.SubmissionRoleFactory(
+        submission=submission_a,
+        submission_id=submission_a.id,
+        user_orcid=logged_in_user.orcid,
+    )
+    db.commit()
+    response = client.request(method="get", url="/api/metadata_submission")
+    assert response.status_code == 200
+
+    results = response.json()["results"]
+    allowed_submission_ids = [result["id"] for result in results]
+    expected_ids = [str(submission_a.id), str(submission_b.id)]
+    assert all([submission_id in expected_ids for submission_id in allowed_submission_ids])
+    assert len(results) == 2
+
+
+@pytest.mark.parametrize(
+    "role,code", [(SubmissionEditorRole.owner, 200), ("author", 200), (None, 403)]
+)
+def test_get_submission_with_roles(
+    db: Session, client: TestClient, token: Token, logged_in_user, role, code
+):
+    if role == "author":
+        submission = fakes.MetadataSubmissionFactory(
+            author=logged_in_user, author_orcid=logged_in_user.orcid
+        )
+        db.commit()
+    elif role == SubmissionEditorRole.owner:
+        submission = fakes.MetadataSubmissionFactory()
+        db.commit()
+        role = fakes.SubmissionRoleFactory(
+            submission=submission, submission_id=submission.id, user_orcid=logged_in_user.orcid
+        )
+    else:
+        submission = fakes.MetadataSubmissionFactory()
+    db.commit()
+    response = client.request(method="get", url=f"/api/metadata_submission/{submission.id}")
+    assert response.status_code == code
+
+
+@pytest.mark.parametrize(
+    "role,code", [(SubmissionEditorRole.owner, 200), ("author", 200), (None, 403)]
```

Comment:
Another consequence (I think) of author directly in the metadata is that the "author" role is not in the enum since it's not to be used in the role table, which is confusing. I wondered a while why one of these was an enum and one a string.

Reply:
If we can ensure that every submission has an `owner` role for its author, we could prevent the scenario that is being tested here (implicit ownership for authors without explicit roles). I agree that this is confusing. Minimally I could add a comment describing the meaning of the test case, but this could also potentially be fixed with a data migration.

Reply:
Yeah, I have the gut feeling that implicit ownership seems less ideal than explicit ownership from a potential bug and maintenance point of view. It's not a huge problem but may make sense to clean it up now rather than later.

Reply:
Tests were fixed up in d7b99f4.

I removed the "author" nonsense, but had to fix some earlier tests by creating explicit roles for the `logged_in_user`. I also updated the `admin_required` dependency (ba8f181) because the tests need to NOT bypass that check (that check was previously only actually done in production environments)